src/components/dev-hub/consistent-nav.js

```diff
@@ -10,8 +12,29 @@ const FrontLayeredNav = styled(UnifiedNav)`
     z-index: ${layer.superFront};
 `;
 
-const ConsistentNav = () => (
-    <FrontLayeredNav position="sticky" property="DEVHUB" />
-);
+const ConsistentNav = () => {
+    const { origin, pathname } = useLocation();
+    /**
+     * There seems to be some race condition where the return link is not being
+     * populated with the correct origin. This tries to delay taking someone
+     * to the wrong place until the origin is loaded.
+     */
+    const returnLink = useMemo(
+        () =>
+            origin
+                ? `${getSsoRegistrationLink(
+                      origin
+                  )}&return_to=${encodeURIComponent(pathname)}`
+                : '#',
+        [origin, pathname]
+    );
```

Comment:
@harikakotipalli This is where the race condition is occurring. I added some additional logic to turn off the link until loaded so perhaps this will do it but if not this is where the issue lies.

Reply:
Noted.Thanks @jestapinski